### PR TITLE
New page header for search relevance

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
New page header for search relevance

### Screenshot
Nav Group Enabled:
Screenshot 2024-08-29 at 7 16 11 PM

Nav Group Disabled
Screenshot 2024-08-29 at 6 43 14 PM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/common/header.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@
  */
 
 import React from 'react';
-import { EuiText, EuiLink, EuiPanel, EuiTitle } from '@elastic/eui';
+import { EuiText, EuiLink, EuiPanel, EuiTitle, EuiSpacer } from '@elastic/eui';
```
Comment (sejli) [inline public/components/common/header.tsx:7]: Is this necessary?

**Review by sejli — COMMENTED** [review]
Might need to run a `yarn test -u` to fix failing CIs.

File: public/components/common/header.tsx (hunk 1 above)
Comment (RyanL1997) [reply public/components/common/header.tsx:7]: ah, that was a left over, reverting. Good catch.

**Comment by RyanL1997** [comment]
The [link checker failure](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/10624361439/job/29452521650?pr=428#step:3:63) seems unrelated to this change:
```
* [ERR] [file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/Using-Docker.md](file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/Using-Docker.md) | Failed: Cannot find file
* [ERR] [file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/Dockerfile](file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/Dockerfile) | Failed: Cannot find file

### Errors in CONTRIBUTING.md

* [ERR] [file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/LICENSE.txt](file:///home/runner/work/dashboards-search-relevance/dashboards-search-relevance/LICENSE.txt) | Failed: Cannot find file
```

**Comment by RyanL1997** [comment]
Adding some fixes according ot the UI feedbacks:
- remove data source picker from top header
- compress data source search bar
- move optional to the right of pipeline "Pipeline - optional" but have it italisized

New Nav Disabled:
Screenshot 2024-08-30 at 5 13 25 PM

New Nav Enabled:
Screenshot 2024-08-30 at 5 12 55 PM

cc: @TackAdam , @kgcreative

**Review by noCharger — APPROVED** [review]
LGTM

**Review by sejli — APPROVED** [review]

**Pull request merged**